Fix bug in Leaf keyword highlighting

fpseverino opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

<!-- Please link to a related issue below. -->
<!-- ie, `Resolves #1234`, etc... so GitHub can magically link it -->

### Changes
Fix a bug in the [Leaf](https://docs.vapor.codes/leaf/overview/) syntax where keywords are highlighted when at the start of a variable name (see the third code block [here](https://docs.vapor.codes/leaf/overview/#template-syntax))

### Checklist
- [x] Added markup tests, or they don't apply here because...
- [x] Updated the changelog at `CHANGES.md`

fpseverino committed:
Fix bug in Leaf keyword highlighting

joshgoebel commented:
Thanks!  Please also add a changelog entry as well as markup test case.

fpseverino committed:
Update CHANGES.md

fpseverino committed:
Update default.txt

fpseverino committed:
Update default.expect.txt

fpseverino commented:
> Please also add a changelog entry as well as markup test case.

@joshgoebel done! I hope I did it correctly

joshgoebel commented:
Except we need a test that would have failed before - so that we prevent this regression in the future... so we need an example of what would have failed before.

joshgoebel committed:
Merge branch 'main' into leaf-bug

joshgoebel commented:
Ah, I see. :)

fpseverino committed:
Merge branch 'main' into leaf-bug

fpseverino committed:
Merge branch 'main' into leaf-bug

fpseverino committed:
Merge branch 'main' into leaf-bug

fpseverino commented:
Hi @joshgoebel, I just wanted to kindly follow up on this PR, just making sure it hasn’t been lost in the shuffle.
Let me know if there’s anything you'd like me to adjust

fpseverino committed:
Merge branch 'main' into leaf-bug

fpseverino commented:
Hi, can we please get this merged?

fpseverino commented:
Hi, just another friendly ping on this PR

joshgoebel commented on test/markup/leaf/default.txt:
```diff
@@ -7,7 +7,7 @@
 
 #if(count(visit) == 1):
     Hello new user!
-#elseif(title == "Welcome back!"):
```
What is this change showing me?

fpseverino replied:
A word like `index` inside a Leaf tag is highlighted incorrectly right now without the fix I'm proposing (see the third code block [here](https://docs.vapor.codes/leaf/overview/#template-syntax)), I changed it so we can keep it tested here

joshgoebel commented on src/languages/leaf.js:
```diff
@@ -26,7 +26,7 @@ export default function(hljs) {
       {
         scope: 'keyword',
-        match: LITERALS.join("|"),
+        match: LITERALS.map(lit => `\\b${lit}\\b`).join("|"),
```
Could we do all this with a single set of `\b`?  Something like:

```suggestion
        match: `\\b${regex.either(LITERALS...)}\\b`)
```

fpseverino replied:
I'm committing this suggestion. I can't test it right now on my machine, so I'll let CI do it

fpseverino committed:
Apply suggestions from code review

Co-authored-by: Josh Goebel <me@joshgoebel.com>

fpseverino committed:
Merge branch 'main' into leaf-bug

fpseverino commented:
Hi @joshgoebel, thanks for the new review! I've addressed your comments, let's see if CI passes and let me know if you need anything else. Pretty excited to finally fix this bug!

fpseverino committed:
Merge branch 'main' into leaf-bug

joshgoebel committed:
Merge branch 'main' into leaf-bug

joshgoebel commented:
I wrote that off the top of my head, looks like it needs work... does the `...` come first?

fpseverino committed:
Fix errors

fpseverino committed:
Merge branch 'main' into leaf-bug

fpseverino commented:
@joshgoebel I think we got it now

joshgoebel committed:
Merge branch 'main' into leaf-bug

joshgoebel commented:
Tests are broken now.

fpseverino committed:
Fix tests

fpseverino committed:
Merge branch 'main' into leaf-bug

fpseverino commented:
> Tests are broken now.

@joshgoebel Sorry about that, I've tested it locally, https://github.com/highlightjs/highlight.js/pull/4248/commits/b13f83bdb6d7ced015882a8d989987ad48e92691 fixes it

joshgoebel committed:
Merge branch 'main' into leaf-bug
